src/Conf.ml

```diff
@@ -856,16 +866,35 @@ let action =
       (List.map !inputs ~f:(fun file ->
            { kind= kind_of file
            ; name= file
-           ; file
+           ; file= (`Input, file)
            ; conf= read_config ~filename:file config } ))
   else
     match !inputs with
+    | ["-"] ->
+        let name =
+          match !name with
+          | None -> impossible "checked by validate"
+          | Some name -> name
+        in
+        let file, oc =
+          Filename.open_temp_file "ocamlformat" (Filename.basename name)
+        in
+        In_channel.iter_lines stdin ~f:(fun s ->
+            Out_channel.output_string oc s ;
+            Out_channel.newline oc ) ;
+        Out_channel.close oc ;
```

Comment:
I thought I made a comment before, but it seems I fumbled it, so sorry for the piecemeal feedback. But could this processing be delayed from here when config options are being read to later in ocamlformat.ml when the actions are being processed? It would be a better separation if actual processing such as copying files wasn't done in the middle of setting up the config. I think it would also probably be cleaner to add an alternative to the action type for processing stdin, and in that way avoid the distinction between Input and Tmp.